addons/payment/views/payment_templates.xml

```diff
@@ -56,14 +56,79 @@
             <t t-set="footer_template_id"
                t-value="footer_template_id or 'payment.footer'"/>
 
+            <!-- === Tokens === -->
+            <t t-if="token_count>0">
+                <div class="mt-3 fs-4">Saved payment methods</div>
+                <div class="card">
+                    <t t-foreach="tokens" t-as="token">
+                        <t t-set="provider" t-value="token.provider_id"/>
+                        <div name="o_payment_option_card" class="card-body o_payment_option_card">
+                            <label>
+                                <!-- === Radio button === -->
+                                <input name="o_payment_radio"
+                                       type="radio"
+                                       t-att-checked="token.id == default_payment_option_id"
+                                       t-att-data-payment-option-id="token.id"
+                                       t-att-data-provider="token.provider_code"
+                                       data-payment-option-type="token"/>
+                                <!-- === Token name === -->
+                                <span class="payment_option_name" t-esc="token.display_name"/>
+                                <!-- === "V" check mark === -->
+                                <t t-call="payment.verified_token_checkmark"/>
+                                <!-- === "Fees" badge === -->
+                                <span t-if="fees_by_provider.get(token.provider_id)"
+                                      class="badge rounded-pill text-bg-secondary ms-1">
+                                        + <t t-esc="fees_by_provider.get(token.provider_id)"
+                                             t-options="{'widget': 'monetary', 'display_currency': currency}"/>
+                                        Fees
+                                </span>
+                                <!-- === "Test token" badge === -->
+                                <t t-call="payment.test_token_badge"/>
+                                <!-- === "Unpublished" badge === -->
+                                <span t-if="not token.provider_id.is_published" class="badge rounded-pill text-bg-danger ms-1">
+                                    Unpublished
+                                </span>
+                                <div class="ms-4">
+                                    <span class="fst-italic small text-muted">
+                                        Managed by
+                                        <!-- It would not work with custom payment provider e.g. 
+                                          "Wire transfer" as it would just show "custom", but
+                                          for now we don't tokenize custom providers, so it's
+                                          not a problem -->
```

Comment:
Good initiative to write your findings down, but it creates more noise than it is useful, IMHO. If someone ever adds tokenization for a custom provider, they will find out that it doesn't work before reading this comment, and they will easily fix the issue anyway.